packages/lesswrong/components/common/LWHomePosts.tsx

```diff
@@ -0,0 +1,368 @@
+
+const settingsButton = (<div className={classes.tagFilterSettingsButton}>
+  <SettingsButton
+    className={classes.hideOnMobile}
+    label={filterSettingsVisibleDesktop ?
+      filterSettingsToggleLabels.desktopVisible :
+      filterSettingsToggleLabels.desktopHidden}
+    showIcon={false}
+    onClick={changeShowTagFilterSettingsDesktop}
+  />
+  <SettingsButton
+    className={classes.hideOnDesktop}
+    showIcon={true}
+    onClick={() => {
+      setFilterSettingsVisibleMobile(!filterSettingsVisibleMobile)
+      captureEvent("filterSettingsClicked", {
+        settingsVisible: !filterSettingsVisibleMobile,
+        settings: filterSettings,
+        pageSectionContext: "latestPosts",
+        mobile: true
+      })
+    }} />
+</div>);
+
+
+  let settings = null;
+
+  if (selectedTab === 'lesswrong-classic') { 
+    settings = <AnalyticsContext pageSectionContext="tagFilterSettings">
+      <div className={classNames({
+        [classes.hideOnDesktop]: !filterSettingsVisibleDesktop,
+        [classes.hideOnMobile]: !filterSettingsVisibleMobile,
+      })}>
+        <TagFilterSettings
+          filterSettings={filterSettings} setPersonalBlogFilter={setPersonalBlogFilter} setTagFilter={setTagFilter} removeTagFilter={removeTagFilter}
+        />
+      </div>
+    </AnalyticsContext>
+  } else if (selectedTab.includes('recombee')) {
+    settings = <div className={classNames({
+      [classes.hideOnDesktop]: !filterSettingsVisibleDesktop,
+      [classes.hideOnMobile]: !filterSettingsVisibleMobile,
+    })}>
+      {userIsAdmin(currentUser) && <RecombeePostsListSettings settings={scenarioConfig} updateSettings={updateScenarioConfig} />}
+    </div>
+  }
+
+
+  const limit = parseInt(query.limit) || defaultLimit;
+  const dateCutoff = moment(now).tz(timezone).subtract(frontpageDaysAgoCutoffSetting.get(), 'days').format("YYYY-MM-DD");
+
+  const recentPostsTerms = {
+    ...query,
+    filterSettings,
+    after: dateCutoff,
+    view: "magic",
+    forum: true,
+    limit:limit
+  };
+
+  return (
+    // TODO: do we need capturePostItemOnMount here?
+    <AnalyticsContext pageSectionContext="postsFeed">
+      <SingleColumnSection>
+        <div className={classes.settingsVisibilityControls}>
+          <TabPicker 
+            sortedTabs={enabledAlgorithms} 
+            defaultTab={selectedTab} 
+            onTabSelectionUpdate={handleSwitchTab}
+            showDescriptionOnHover
+          />
+          {showSettingsButton && settingsButton}
+        </div>
+        {settings}
+        {isFriendlyUI && <StickiedPosts />}
+        {/* TODO: reenable, disabled for testing to see how often duplication happens */}
+        <HideRepeatedPostsProvider>
+          <AnalyticsContext listContext={"latestPosts"}>
+            {/* Allow hiding posts from the front page*/}
+            <AllowHidingFrontPagePostsContext.Provider value={true}>
+
+              {/* LATEST POSTS (Hacker News Algorithm) */}
```

Comment:
Surely, if we're always wrapping whatever posts we're showing in `<AnalyticsContext feedType={selectedTab}>`, we should not repeat that for every case?